bears/general/InvalidLinkBear.py

```diff
@@ -211,3 +220,6 @@ def run(self, filename, file,
                             file=filename,
                             line=line_number,
                             severity=RESULT_SEVERITY.NORMAL)
+
+        # Clear the status code cache after the file has been analyzed
+        InvalidLinkBear.clear_status_code_cache()
```

Comment:
could we just clear the cache in `__del__`? It's not particularly good style but it'd work and if it doesn't no real harm is done.